### DISCUSSION TITLE
Vehicle sheet

### DISCUSSION THREAD

**Discussion by QckSilverDragon** [Muttley/foundryvtt-fallout#521] [Show and tell]
So i decided to try making the vehicle sheet. For a start i decided to take inspiration from the creature sheet.

I debated on how to mange the vehicle qualities and thought the weapon qualities would be nice.
After a brain fart on my part i got that working. I just need to get the cargo quality to increase carry limit and make sure the inventory works.

TODO:
**Cargo**: Need to calculate cargo and be modified by Cargo quality.

**Body locations**: I plan to use the dropdown to select the different vehicle types but there are 3 or 4 different configurations so yay. 
For body hit location, i think i want to add a button to roll for it manually. Adding vehicle hit locations to the existing attack rolls would make it insanely long. I added DC dice rollers to the impact and cover fields so this should be easy.

**Weapons**: I want to make it so attack rolls use the users character sheet for the roll.

![image](https://github.com/user-attachments/assets/19bf3f25-a54e-4fd8-bd59-a2ab02bb7169)

**Comment by QckSilverDragon** [comment]
@Muttley I do have a question about the i18n files. Do all the languages need to be updated or just the english one?

**Reply by Muttley** [reply]
Just the English file.  That's the master and all others are handled through Crowdin

**Comment by Muttley** [comment]
Looking good so far.  :)

**Comment by QckSilverDragon** [comment]
Im happy that your robust prep made this integrate easily. Now to just remove the poison and radiation resistances.

![body](https://github.com/user-attachments/assets/231d7d1b-d828-484d-94ce-062fb30215bd)

**Comment by QckSilverDragon** [comment]
More progress.

Abilities tab
![image](https://github.com/user-attachments/assets/3a217b59-2d00-4c55-b902-f0d16c3df333)

Weapons tab. Thinking of adding it below the cargo tab. Just wanted people to be able to "favorite" equipped weapons.
![image](https://github.com/user-attachments/assets/5f6a5981-0a08-4d2d-81b3-67184e2a38e2)

Cargo tab
![image](https://github.com/user-attachments/assets/3b119105-89d5-49fa-9396-08c7b6185c8c)

**Reply by Muttley** [reply]
> More progress.
> 
> Weapons tab. Thinking of adding it below the cargo tab. Just wanted people to be able to "favorite" equipped weapons. ![image](https://private-user-images.githubusercontent.com/4044050/391196647-5f6a5981-0a08-4d2d-81b3-67184e2a38e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMwNzE0MDEsIm5iZiI6MTczMzA3MTEwMSwicGF0aCI6Ii80MDQ0MDUwLzM5MTE5NjY0Ny01ZjZhNTk4MS0wYTA4LTRkMmQtODFiMy02NzE4NGUyYTM4ZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDFUMTYzODIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjNjNzU4ZjJjNDk0Y2ZjNTQ0MDRkZTk2OWJmZTJlZTFhMDY5NmJhNTZlMDcyNTQzZjFiM2I2ZjA5OGIzMWE2MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.CMEzfxhS-qHkiJlEVUiXS3kYbHF9PmsLNaa9VH53JmQ)

I've not read up on the Vehicles, but do the weapons use attributes/skills from whoever's using the vehicle?  If so we'd need a way to link the driver to the vehicle (probably in a similar way to the way the Settlement leader is selected, but just character's who have a player owner).

**Reply by QckSilverDragon** [reply]
Went over the rules again and realized i removed injury locations, thinking they weren't needed but they are a thing. So i need to fix that.

As for weapons, "Vehicle’s gunners can make attacks with mounted weapons in the same way as a personal weapon, using whatever skill applies to the weapon".

I debated on adding characters to "positions" in the vehicle but couldn't figure out a way to do it without having to make custom sheets for all. Maybe do something similar to the settlements list but also have some drop down or something to select position. 

In the end i decided the easiest thing to do is to make it so the user can click the weapon in the vehicle and it will use the stats of the assigned character sheet for this user. I copied and tweaked your function for getting actors for macros. This way the GM could select a token to make the attack with or it just gives a blank 2d20 roller and they fill in the numbers.

P.S. the ammo will be deducted from the vehicle and not the user.

**Reply by Muttley** [reply]
Yeah, that should work as a first step.  We can always revisit it later if needed.

**Reply by QckSilverDragon** [reply]
Also, im not an artist so i didnt make a new default image for the vehicle sheet. Was going to have to leave that to you.

**Reply by Muttley** [reply]
> Also, im not an artist so i didnt make a new default image for the vehicle sheet. Was going to have to leave that to you.

Hah, nobody wants to see my artwork.  Why do you think there aren't alternative images for the different robot body types.  ;)

**Comment by QckSilverDragon** [comment]
@Muttley i think ill be done with this soon. Should i just do a pull request or do you want to check it some other way first?

**Reply by Muttley** [reply]
Yeah, do a pull request when you're ready and I'll check it then.  Make sure to pull the latest changes from my develop branch before you do and resolve any conflicts, hopefully there won't be any though.

**Reply by QckSilverDragon** [reply]
The only issue in the last update was your breaking up of the NPC sheets. I had a decent amount of code in these files also and had to separate out all my vehicle stuff before i merged.

**Reply by Muttley** [reply]
> The only issue in the last update was your breaking up of the NPC sheets. I had a decent amount of code in these files also and had to separate out all my vehicle stuff before i merged.

Yeah, that was unlucky timing, but the templates really needed to be tidied up.

**Reply by QckSilverDragon** [reply]
Now that i completed my issue request. :) Guess i should try and work on other peoples requests.
